ambient-peer/README.md

```diff
@@ -0,0 +1,109 @@
+# Ambient peer discovery
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2023-10-16  |
+
+Authors: [@thomaseizinger]
+
+Interest Group: <!-- Please add yourself here. -->
+
+[@thomaseizinger]: https://github.com/thomaseizinger
+
+See the [lifecycle document][lifecycle-spec] for context about the maturity level and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Table of Contents
+
+<!-- TODO -->
+
+## Overview
+
+The ambient peer discovery protocol allows peers to share some of their ambient peers with each other.
+
+## Usecase
+
+Ambient peer discovery is most useful when a node either starts with or is left with a few or perhaps only a single connection.
+
+For example, a user may start a libp2p web app and enter another browser's relayed `/webrtc` address.
+The connection will succeed but because both nodes are browsers, further discovery of nodes via e.g. kademlia is not possible.
+Ambient peer discovery allows the web app to inquire for further nodes from the new connection.
+
+The protocol is designed to compliment other discovery mechanism like kademlia.
+It features a very small resource footprint and can thus also be used by lite-clients within browser or mobile environments.
+
+## Protocol
+
+1. Node _A_ opens a new stream to node _B_ with the protocol name `/libp2p/ambient-peers`.
+1. Node _B_ chooses a subset of at most 5 known peer records received from other peers.
+   1. The chosen peer records SHOULD at least have one address that share the same transport technology as the the connection between node _A_ and node _B_.
+      For example, if node _A_ and node _B_ are connected via WebRTC, node _B_ SHOULD select 5 peer records where each one of them has at least one WebRTC address.
+   1. Node _B_ SHOULD NOT be currently connected to any of these nodes.
+1. Node _B_ writes these peer records onto the stream in their [protobuf encoding](https://github.com/libp2p/specs/blob/master/RFC/0003-routing-records.md#address-record-format), each record being length-prefixed using an unsigned varint and closes the stream after the last one.
+1. Node _A_ reads peer records from the stream until EOF or 5 have been received, whichever comes earlier.
+
+## Security considerations
+
+Revealing even just some of your peers has serious privacy and security implications for a network.
+By default, implementations MUST NOT share records of peers they are currently connected to.
+Implementations MAY add a configuration flag that allows users to override this.
+
+<!-- @vyzo to add more text here -->
+
+## Implementation considerations
+
+### Bound local peer storage
+
+This protocol requires nodes to store records of peers they used to be connected to.
+This is useful independently of this protocol to e.g. reconnect to a peer you've once been connected to.
```

Comment:
This adds a lot of complexity to implementations.

Reply:
Can you elaborate on why this adds complexity? I appreciate that the various implementations differ greatly in their internal design but fundamentally, this can easily be satisfied with an LRU cache. Really, all we are doing is accessing the ["peer store"](https://docs.libp2p.io/concepts/fundamentals/peers/#peer-store). I can make that clearer if you want?